ports/vcpkg-make/vcpkg_make_common.cmake

```
@@ -0,0 +1,657 @@
+include_guard(GLOBAL)
+
+### Mapping variables
+macro(z_vcpkg_make_set_common_vars)
+    set(path_suffix_RELEASE "")
+    set(path_suffix_DEBUG "/debug")
+    set(suffix_RELEASE "rel")
+    set(suffix_DEBUG "dbg")
+    foreach(config IN ITEMS RELEASE DEBUG)
+        set("workdir_${config}" "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-${suffix_${config}}")
+    endforeach()
+    set(buildtypes release)
+    if(NOT VCPKG_BUILD_TYPE)
+        list(APPEND buildtypes debug)
+    endif()
+endmacro()
+
+###
+macro(z_vcpkg_make_get_cmake_vars)
+    cmake_parse_arguments(vmgcv_arg # Not just arg since macros don't define their own var scope. 
+        "" "" "LANGUAGES" ${ARGN}
+    )
+
+    z_vcpkg_get_global_property(has_cmake_vars_file "make_cmake_vars_file" SET)
+
+    if(NOT has_cmake_vars_file)
+        if(vmgcv_arg_LANGUAGES)
+            # Escape semicolons to prevent CMake from splitting LANGUAGES list when passing as -D option.
+            string(REPLACE ";" "\;" vmgcv_arg_langs "${vmgcv_arg_LANGUAGES}")
+            list(APPEND VCPKG_CMAKE_CONFIGURE_OPTIONS "-DVCPKG_LANGUAGES=${vmgcv_arg_langs}")
+            unset(langs)
+        endif()
+
+        list(APPEND VCPKG_CMAKE_CONFIGURE_OPTIONS "-DVCPKG_DEFAULT_VARS_TO_CHECK=CMAKE_LIBRARY_PATH_FLAG")
+        vcpkg_cmake_get_vars(cmake_vars_file)
+        z_vcpkg_set_global_property(make_cmake_vars_file "${cmake_vars_file}")
+    else()
+        z_vcpkg_get_global_property(cmake_vars_file "make_cmake_vars_file")
+    endif()
+
+    include("${cmake_vars_file}")
+endmacro()
+
+### normalize architectures
+function(z_vcpkg_make_determine_arch out_var value)
+    if(${value} MATCHES "^(amd|AMD|x)64$")
+        set(${out_var} x86_64 PARENT_SCOPE)
+    elseif(${value} MATCHES "^(x|X)86$")
+        set(${out_var} i686 PARENT_SCOPE)
+    elseif(${value} MATCHES "^(ARM|arm)64$")
+        set(${out_var} aarch64 PARENT_SCOPE)
+    elseif(${value} MATCHES "^(ARM|arm)$")
+        set(${out_var} arm PARENT_SCOPE)
+    elseif(${value} MATCHES "^(x86_64|i686|aarch64|arm)$" OR NOT VCPKG_TARGET_IS_WINDOWS)

```

```suggestion
    elseif(${value} MATCHES "^(x86_64|i686|aarch64)$" OR NOT VCPKG_TARGET_IS_WINDOWS)
```
I think the previous conditional checks this case?